### ISSUE TITLE
Remove all notes example from the README not working

### ISSUE THREAD

**Issue by Phibonacci** [melanchall/drywetmidi#256]
I am looking to remove all the notes from each chords but the first one, for each track. To get started I tried the following example from the README that is supposed to remove absolutely all notes from the midi.

```cs
using Melanchall.DryWetMidi.Core;
using Melanchall.DryWetMidi.Interaction;

var midiFile = MidiFile.Read("myfile.mid");
foreach (var trackChunk in midiFile.GetTrackChunks())
{
    using (var notesManager = trackChunk.ManageNotes())
    {
        notesManager.Notes.RemoveAll(n => n.NoteName == NoteName.CSharp);
    }
}
```

`notesManager.Notes` does not exist.

> error CS1061: 'TimedObjectsManager<Note>' does not contain a definition for 'Notes' and no accessible extension method 'Notes' accepting a first argument of type 'TimedObjectsManager<Note>' could be found (are you missing a using directive or an assembly reference?)

**Comment by melanchall** [comment]
Hi,

Thank you, I'll fix the example. You just need to use `notesManager.Objects`.

**Comment by Phibonacci** [comment]
Thanks for your quick answer.

I looked at the `ManageNotes` extension but then I get an `IEnumerable` that cannot be modified.

**Comment by melanchall** [comment]
> I looked at the ManageNotes extension but then I get an IEnumerable that cannot be modified.

Where? What code? Please provide details.

**Comment by Phibonacci** [comment]
After going through the library I found a method to update the chords. I thought this would allow me to remove all notes from each chord but the first one, on all tracks.

```cs
using Melanchall.DryWetMidi.Core;
using Melanchall.DryWetMidi.Interaction;

var midiFile = MidiFile.Read("in.mid");
foreach (var trackChunk in midiFile.GetTrackChunks())
{
    trackChunk.ProcessChords(TrimChord);
}
midiFile.Write("out.mid");

void TrimChord(Chord chord)
{
    chord.Notes.RemoveAll(x => x != chord.Notes.First());
}
```

But the output file is identical to the input one.

**Comment by Phibonacci** [comment]
This worked.

```cs
using Melanchall.DryWetMidi.Core;
using Melanchall.DryWetMidi.Interaction;

var midiFile = MidiFile.Read("in.mid");
foreach (var trackChunk in midiFile.GetTrackChunks())
{
    var newChords = new List<Chord>();
    foreach (var chord in trackChunk.GetChords())
    {
        chord.Notes.RemoveAll(x => x != chord.Notes.First());
        newChords.Add(chord);
    }
    trackChunk.RemoveChords();
    trackChunk.AddObjects(newChords);
}
midiFile.Write("out.mid");
```

**Comment by melanchall** [comment]
> I'm guessing the ProcessChords is making copies of the chords. How can I edit them instead?

That's correct. But the file should be updated. Hmm, it's probably a bug in the library. I'll check it. Please don't close the issue.

**Comment by melanchall** [comment]
Yes, that's the bug. I'll think howto support chord's notes collection changes within `ProcessChords` methods.

**Comment by melanchall** [comment]
I've released [DryWetMIDI 7.0.0-prerelease2](https://www.nuget.org/packages/Melanchall.DryWetMidi/7.0.0-prerelease2) which introduces `ChordProcessingHint` parameter in `ProcessChords` methods. So now for your task you can use this code:

```csharp
midiFile.ProcessChords(
    chord => 
    {
        var higherNote = FindHigherNote(chord.Notes);
        chord.Notes.RemoveAll(x => x != higherNote);
    },
    hint: ChordProcessingHint.NotesCollectionCanBeChanged);
```

Thanks for the issue!

**Comment by melanchall** [comment]
:rocket: **7.0.0 version is released now!**

Prerelease NuGet packages will be unlisted soon, so please update the package references to the new version.
